fms_parser.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/PuerkitoBio/goquery"
+	"os"
+	"strings"
+)
+
+// func main() {
+// 	val := ParseHTMLtoJSON("raw0.html")
+// 	fmt.Println(string(val))
+// }
+
+func split_and_strip(text string, separator string) ([]string) {
+	// Split text into parts at separator character. Remove whitespace from parts.
+	// "a • b • c" -> ["a", "b", "c"]
+	parts := strings.Split(text, separator)
+	var result []string
+	for _, part := range parts {
+		result = append(result, strings.TrimSpace(part))
+	}
+	return result
+}
+
+func ParseHTMLtoJSON(filename string) ([]byte) {
+	//////////////////////////////////////////////////
+	// Parse html from FMS into TBA-compatible JSON //
+	//////////////////////////////////////////////////
+
+	// Open file
+	r, err := os.Open(filename)
+	if err != nil {
+		fmt.Println("Error opening file", filename)
```

Comment:
You should make this function return an error instead, probably as a second return value.